optimum/onnxruntime/modeling_seq2seq.py

```diff
@@ -198,474 +201,160 @@
 """
 
 
-class ORTModelForConditionalGeneration(ORTModel, ABC):
+class ORTEncoder:
     """
-    Sequence-to-sequence model with a language modeling head for ONNX Runtime inference.
-
-    Important attributes:
-        config ([`PretrainedConfig`]):
-            Instance of the configuration associated to the model. Initializing with a config file does
-            not load the weights associated with the model, only the configuration.
-        use_io_binding (`bool`):
-            Whether use IOBinding during inference to avoid memory copy between the host and devices. Defaults to `True`
-            if the device is CUDA, otherwise defaults to `False`.
-        use_cache (`bool`):
-            Whether or not past key/values cache should be used. It is determined by whether an InferenceSession for
-            that was provided or not.
-        providers (`List[str`]):
-            The list of execution providers the model is running on.
-        encoder (`ORTEncoder`):
-            The encoder model.
-        decoder (`ORTDecoder`):
-            The decoder model.
-        decoder_with_past (`Optional[ORTDecoder]`):
-            The decoder model handling the past key/values if `use_cache=True`, else `None`.
-
-    Other attributes:
-        encoder_file_name (`str`, defaults to `optimum.onnxruntime.utils.ONNX_ENCODER_NAME`):
-            The name of the ONNX file containing the encoder part of the model.
-        decoder_file_name (`str`,  defaults to `optimum.onnxruntime.utils.ONNX_DECODER_NAME`):
-            The name of the ONNX file containing the decoder part of the model.
-        decoder_file_with_past_name (`str`, defaults to `optimum.onnxruntime.utils.ONNX_DECODER_WITH_PAST_NAME`):
-            The name of the ONNX file containing the decoder with past key/values part of the model.
-        model_save_dir (`str`, defaults to `""`):
-            The directory under which the model exported to ONNX was saved.
+    Encoder model for ONNX Runtime inference.
 
+    Args:
+        session (`ort.InferenceSession`):
+            The ONNX Runtime inference session associated to the encoder.
```

Comment:
Why only `session`? The `__init__` seems to have more arguments.
I also think the `Args: ...` part should be moved to the `__init__`'s docstring, even though I'm not sure there is any consistency regarding this in HF libraries.

Reply:
You are right!